addressing/README.md

```diff
@@ -0,0 +1,329 @@
+
+In a system that only supported TCP/IP or UDP over IP, we could easily write our
+addresses with the familiar `<ip>:<port>` notation and store them as tuples of
+address and port. However, libp2p was designed to be transport agnostic, which
+means that we can't assume that we'll even be using an IP-backed network at all.
+
+To support a growing set of transport protocols without special-casing each
+addressing scheme, libp2p uses [multiaddr][multiaddr-repo] to encode network
+addresses for all supported transport protocols.
+
+This document covers [how multiaddr is used in libp2p](#multiaddr-in-libp2p).
+For more information on other use cases, or to find links to multiaddr
+implementations in various languages, see the [mulitaddr
+repository][multiaddr-repo].
+
+## multiaddr in libp2p
+
+multiaddr is used throughout libp2p for encoding network addresses, and
+addresses are generally exchanged over the wire as binary-encoded multiaddrs in
+libp2p's core protocols. 
+
+When exchanging addresses, peers send a multiaddr containing both their network
+address and peer id, as described in [the section on the `p2p`
+multiaddr](#the-p2p-multiaddr).
+
+### multiaddr basics
+
+A multiaddr generally represents a path through a stack of successively
+"higher-level" protocols that can be traversed to some destination.
+
+For example, the `/ip4/7.7.7.7/tcp/1234` multiaddr starts with `ip4`, which is
+the lowest-level protocol that requires an address. The `tcp` protocol is
+encapsulated within `ip4`, so it comes next.
+
+The multiaddr above consists of two components, the `/ip4/7.7.7.7` component,
+and the `/tcp/1234` component. It's not possible to split either one further;
+`/ip4` alone is an invalid multiaddr, because the `ip4` protocol was defined to
+require a 32 bit address. Similarly, `tcp` requires a 16 bit port number.
+
+Although we referred to `/ip4/7.7.7.7` and `/tcp/1234` as "components" of a
+larger TCP/IP address, each is actually a valid multiaddr according to the
+multiaddr spec. However, not every valid multiaddr describes a complete path
+through the network. As we've seen, even a simple TCP/IP connection requires
+composing two multiaddrs into one. See the section on [composing
+multiaddrs](#composing-multiaddrs) for information on how multiaddrs can be
+combined, and the [Transport multiaddrs section](#transport-multiaddrs) for the
+combinations that describe valid transport addresses.
+
+The [multiaddr protocol table][multiaddr-proto-table] contains all currently
+defined protocols and the length of their address components.
+
+Some multiaddr protocols do not require any additional addressing information.
+For example, WebSockets are by definition encapsulated within TCP/IP, so the
+`/ws` multiaddr protocol is encapsulated within a TCP/IP multiaddr:
+`/ip4/7.7.7.7/tcp/1234/ws`. This address is composed of three distinct
+multiaddrs, `/ip4/7.7.7.7`, `/tcp/1234`, and `/ws`.
```

Comment:
The term multiaddr is overloaded here. I prefer to promote the term "multiaddr" to imply "sound/functional multiaddr" within this spec. All of the above should be non-normative narrative, and this spec should focus on the technical norms to form "sound/functional multiaddrs" that libp2p implementations should understand.

Reply:
Fixed in f3df677e4101fe266d1f806c58607790751aa6c9.